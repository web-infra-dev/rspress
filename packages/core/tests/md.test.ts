import rehypeStringify from 'rehype-stringify';
import remarkParse from 'remark-parse';
import remarkRehype from 'remark-rehype';
import { unified } from 'unified';
import { describe, expect, test } from 'vitest';

describe('Markdown compile cases', () => {
  const processor = unified()
    .use(remarkParse)
    .use(remarkRehype)
    .use(rehypeStringify);

  test('Compile title', async () => {
    const mdContent = '# 123';
    const result = processor.processSync(mdContent);
    expect(result.value).toMatchInlineSnapshot('"<h1>123</h1>"');
  });
});
