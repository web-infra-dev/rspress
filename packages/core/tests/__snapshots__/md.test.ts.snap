// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Markdown compile cases > remarkPluginNormalizeLink > should just work 1`] = `
"import image0 from "./test3.jpg"

import image1 from "./test4.png"

[link1](/test1.html)

{/* jsx link will not be transformed */}

<a href="./test2">link2</a>

<img alt="alt1" src={image0} />

<img src={image1} alt="alt2" />
"
`;
